src/T2DMap.cpp

```diff
@@ -4110,9 +4133,12 @@ void T2DMap::slot_setArea()
                     }
                 }
             }
+            auto& targetAreaName = areaNamesMap.value(newAreaId);
+            mpMap->mpMapper->comboBox_showArea->setCurrentText(targetAreaName);
+            slot_switchArea(targetAreaName);
```

Comment:
:warning: This code is version sensitive it will only compile with Qt versions ***less than* 5.15.0** as the method is NOT a slot in the later versions!
```suggestion
#if (QT_VERSION) >= (QT_VERSION_CHECK(5, 15, 0))
            switchArea(targetAreaName);
#else
            slot_switchArea(targetAreaName);
#endif
```